bwe/RateController.cpp

```diff
@@ -298,12 +327,24 @@ RateController::BacklogAnalysis RateController::analyzeBacklog(uint32_t reportNt
             item.delaySinceSR > ntp32Second / 20)
         {
             auto reportCandidate = analyzeProbe(i, modelBandwidthKbps);
```

Comment:
```suggestion
     auto reportCandidate = analyzeProbe(i, modelBandwidthKbps);
     if (reportCandidate.senderReportItem->reportNtp == _lastProbe.ntp &&
        reportCandidate.delaySinceSR == _lastProbe.delaySinceSR &&
        reportCandidate.packetsReceived == _lastProbe.packetsReceived)
    {
       // From now on we are only wasting CP and probably ignoring a good recent by choosing one that already was
       // processed
        return report;
    }

```

Reply:
I think here we are potential ignore newer candidates if we find a older one that will be ignored because it was already utilised. Maybe we should mark a report as analysed and ignore it

Imagine in backlog I have:
[report2]
[report1]


and we decide that report1 is better than 2. Then we use the 1.

Next I have:
[report3]
[report2]
[report1]
 
And I decide to choose again the 1. But the report3 is more recent and we already have used the 1. Shouldn't we use the 3?

Reply:
For me also make sense mark the 2 as analysed. Because if we prefer the report 1 over 2, doesn't make sense in next iteration use the report 2 just because the 1 is not in the backlog any longer

Reply:
I agree, I was also thinking about marking which reports we already used. A problem with that is that the report A may be the best atm, then another B has more confirmed packets, then another RB confirms more packets and the A is actually the best again. But we cannot use A because it was marked as used.

Reply:
I am tempted to add the improvement you suggest above. The rationale would be that if a later RB comes in, it should confirm packets also in that last best reports and other reports. We would then recalculate all reports and pick the best.

But what if _lastProbe does not contain any packets from the RB ssrc? What if it is an RB for probe packets that are important but sits right after a previous SR?
We would not look beyond the _lastProbe then. Maybe it is unusual but we need to resolve that.
It could be done when analysing the RB. That we clear a mark or something in lastProbe if the RB affects any packets before lastProbe.